presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/AbstractResourceConfigurationManager.java

```diff
@@ -98,13 +98,17 @@ private void validateSelectors(List<ResourceGroupSpec> groups, List<ResourceGrou
     protected AbstractResourceConfigurationManager(ClusterMemoryPoolManager memoryPoolManager)
     {
         memoryPoolManager.addChangeListener(new MemoryPoolId("general"), poolInfo -> {
+            Map<ResourceGroup, DataSize> memoryLimits = new HashMap<>();
```

Comment:
maybe we can add a comment here of why we put the limits in a map and then apply.